src/parquet/file/reader-internal.cc

```diff
@@ -171,16 +171,37 @@ std::unique_ptr<PageReader> SerializedRowGroup::GetColumnPageReader(int i) {
       std::move(stream), FromThrift(col.meta_data.codec), properties_.allocator()));
 }
 
-RowGroupStatistics SerializedRowGroup::GetColumnStats(int i) const {
-  const format::ColumnMetaData& meta_data = metadata_->columns[i].meta_data;
+template <typename DType>
+static std::shared_ptr<RowGroupStatistics> MakeColumnStats(
+    const format::ColumnMetaData& metadata, const ColumnDescriptor* descr) {
+  return std::make_shared<TypedRowGroupStatistics<DType>>(descr, metadata.statistics.min,
+      metadata.statistics.max, metadata.num_values, metadata.statistics.null_count,
+      metadata.statistics.distinct_count);
+}
 
-  RowGroupStatistics result;
-  result.num_values = meta_data.num_values;
-  result.null_count = meta_data.statistics.null_count;
-  result.distinct_count = meta_data.statistics.distinct_count;
-  result.max = &meta_data.statistics.max;
-  result.min = &meta_data.statistics.min;
-  return result;
+std::shared_ptr<RowGroupStatistics> SerializedRowGroup::GetColumnStats(int i) const {
+  const format::ColumnMetaData& meta_data = metadata_->columns[i].meta_data;
+  const ColumnDescriptor* descr = schema_->Column(i);
+
+  switch (meta_data.type) {
+    case Type::BOOLEAN:
+      return MakeColumnStats<DataType<Type::BOOLEAN>>(meta_data, descr);
+    case Type::INT32:
+      return MakeColumnStats<DataType<Type::INT32>>(meta_data, descr);
+    case Type::INT64:
+      return MakeColumnStats<DataType<Type::INT64>>(meta_data, descr);
+    case Type::INT96:
+      return MakeColumnStats<DataType<Type::INT96>>(meta_data, descr);
+    case Type::DOUBLE:
+      return MakeColumnStats<DataType<Type::DOUBLE>>(meta_data, descr);
+    case Type::FLOAT:
+      return MakeColumnStats<DataType<Type::FLOAT>>(meta_data, descr);
+    case Type::BYTE_ARRAY:
+      return MakeColumnStats<DataType<Type::BYTE_ARRAY>>(meta_data, descr);
+    case Type::FIXED_LEN_BYTE_ARRAY:
+      return MakeColumnStats<DataType<Type::FIXED_LEN_BYTE_ARRAY>>(meta_data, descr);
```

Comment:
Use the typedefs for these `DataType<T>` values? I agree it might be nicer to move this up into `TypedColumnReader` somehow and skip the switch statement, but not necessary at the moment